comp/checks/winregistry/impl/integration_logs_registry_delegate.go

```diff
@@ -0,0 +1,120 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build windows
+
+package winregistryimpl
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/comp/logs/agent"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	"time"
+)
+
+type integrationLogsRegistryDelegate struct {
+	baseRegistryDelegate
+	logsComponent agent.Component
+	valueMap      map[string]interface{}
+	origin        *message.Origin
+	muted         bool
+}
+
+const (
+	keyChanged = "key_changed"
+	keyCreated = "key_created"
+	keyDeleted = "key_deleted"
+)
+
+func (i *integrationLogsRegistryDelegate) sendLog(payload message.StructuredContent, status string) {
+	if i.logsComponent == nil || i.muted {
+		return
+	}
+	msg := message.NewStructuredMessage(payload, i.origin, status, time.Now().UnixNano())
+	i.logsComponent.GetPipelineProvider().NextPipelineChan() <- msg
+}
+
+func getPayload(m, eventType string, regKeyCfg registryKey) message.BasicStructuredContent {
+	payload := message.BasicStructuredContent{
+		Data: make(map[string]interface{}),
+	}
+	payload.Data["message"] = m
+	payload.Data["type"] = "windows_registry_event"
+	payload.Data["key_path"] = regKeyCfg.originalKeyPath
+	payload.Data["event_type"] = eventType
+
+	//payload.Data["process_id"] = coming in a future version
+	//payload.Data["process_name"] = coming in a future version
+	//payload.Data["thread_id"] = coming in a future version
+	return payload
+}
+
+func getKeyName(valueName string, regKeyCfg registryKey) string {
+	return fmt.Sprintf("%s\\%s", regKeyCfg.originalKeyPath, valueName)
+}
+
+func (i *integrationLogsRegistryDelegate) processValue(valueName string, val interface{}, regKeyCfg registryKey) {
+	keyName := getKeyName(valueName, regKeyCfg)
+	cachedVal, ok := i.valueMap[keyName]
+	if !ok {
+		p := getPayload(fmt.Sprintf("value %s = '%v'", keyName, val), keyCreated, regKeyCfg)
+		p.Data["old_value"] = nil
+		p.Data["new_value"] = val
+		i.sendLog(&p, "info")
+		i.valueMap[keyName] = val
+	} else {
+		if cachedVal != val {
```

Comment:
I feel that there is a room for a problem here. It is a bit contrived but I feel that in some cases the type of the value may not match if "old" value, e.g. is integer but had been deleted and recreated as a string between two subsequent check. In result cachedVal would be integer but val as a string. Not sure what will happen if they are compared.